src/main/scala/util/MultiWidthFifo.scala

```diff
@@ -23,81 +24,81 @@ class MultiWidthFifo(inW: Int, outW: Int, n: Int) extends Module {
     require(inW % outW == 0, s"MultiWidthFifo: in: $inW not divisible by out: $outW")
     require(n % nBeats == 0, s"Cannot store $n output words when output beats is $nBeats")
 
-    val wdata = Reg(Vec(n / nBeats, Bits(width = inW)))
-    val rdata = Vec(wdata.flatMap { indat =>
+    val wdata = Reg(Vec(n / nBeats, Bits(inW.W)))
+    val rdata = VecInit(wdata.flatMap { indat =>
       (0 until nBeats).map(i => indat(outW * (i + 1) - 1, outW * i)) })
 
-    val head = Reg(init = UInt(0, log2Up(n / nBeats)))
-    val tail = Reg(init = UInt(0, log2Up(n)))
-    val size = Reg(init = UInt(0, log2Up(n + 1)))
+    val head = RegInit(0.U(log2Up(n / nBeats).W))
+    val tail = RegInit(0.U(log2Up(n).W))
+    val size = RegInit(0.U(log2Up(n + 1).W))
 
-    when (io.in.fire()) {
+    when (io.in.fire) {
       wdata(head) := io.in.bits
-      head := head + UInt(1)
+      head := head + 1.U
     }
 
-    when (io.out.fire()) { tail := tail + UInt(1) }
+    when (io.out.fire) { tail := tail + 1.U }
 
     size := MuxCase(size, Seq(
-      (io.in.fire() && io.out.fire()) -> (size + UInt(nBeats - 1)),
-      io.in.fire() -> (size + UInt(nBeats)),
-      io.out.fire() -> (size - UInt(1))))
+      (io.in.fire && io.out.fire) -> (size + (nBeats - 1).U),
+      io.in.fire -> (size + nBeats.U),
+      io.out.fire -> (size - 1.U)))
 
-    io.out.valid := size > UInt(0)
+    io.out.valid := size > 0.U
     io.out.bits := rdata(tail)
-    io.in.ready := size < UInt(n)
+    io.in.ready := size < n.U
     io.count := size
   } else {
     val nBeats = outW / inW
 
     require(outW % inW == 0, s"MultiWidthFifo: out: $outW not divisible by in: $inW")
 
-    val wdata = Reg(Vec(n * nBeats, Bits(width = inW)))
-    val rdata = Vec.tabulate(n) { i =>
+    val wdata = Reg(Vec(n * nBeats, Bits(inW.W)))
+    val rdata = VecInit.tabulate(n) { i =>
       Cat(wdata.slice(i * nBeats, (i + 1) * nBeats).reverse)}
 
-    val head = Reg(init = UInt(0, log2Up(n * nBeats)))
-    val tail = Reg(init = UInt(0, log2Up(n)))
-    val size = Reg(init = UInt(0, log2Up(n * nBeats + 1)))
+    val head = RegInit(0.U(log2Up(n * nBeats).W))
+    val tail = RegInit(0.U(log2Up(n).W))
+    val size = RegInit(0.U(log2Up(n * nBeats + 1).W))
 
-    when (io.in.fire()) {
+    when (io.in.fire) {
       wdata(head) := io.in.bits
-      head := head + UInt(1)
+      head := head + 1.U
     }
 
-    when (io.out.fire()) { tail := tail + UInt(1) }
+    when (io.out.fire) { tail := tail + 1.U }
 
     size := MuxCase(size, Seq(
-      (io.in.fire() && io.out.fire()) -> (size - UInt(nBeats - 1)),
-      io.in.fire() -> (size + UInt(1)),
-      io.out.fire() -> (size - UInt(nBeats))))
+      (io.in.fire && io.out.fire) -> (size - (nBeats - 1).U),
+      io.in.fire -> (size + 1.U),
+      io.out.fire -> (size - nBeats.U)))
 
-    io.count := size >> UInt(log2Up(nBeats))
-    io.out.valid := io.count > UInt(0)
+    io.count := size >> log2Up(nBeats)
+    io.out.valid := io.count > 0.U
```

Comment:
why no `UInt` here

Reply:
Because we are shifting by a fixed amount. These two should be equivalent in generated FIRRTL?